compression/zlib/src/main/java/net/daporkchop/lib/compression/zlib/java/JavaZlib.java

```diff
@@ -66,7 +67,8 @@ public ZlibDeflater deflater(int level, int strategy, int mode) {
 
     @Override
     public ZlibInflater inflater(int mode) {
-        throw new UnsupportedOperationException(); //TODO
+        checkArg(mode != Zlib.MODE_GZIP, "gzip mode not supported!");
+        return new JavaZlibInflater(this, mode == Zlib.MODE_RAW);
```

Comment:
please don't copy this gross hack over from SaveSearcher... it doesn't support all the features it's supposed to. either implement pure-java zlib properly, or leave it entirely unsupported so i can implement it properly myself later on.